### DISCUSSION TITLE
Question: filename of corrected reads

### DISCUSSION THREAD

**Discussion by bernt-matthias** [ablab/spades#1078] [Q&A] [answered]
### Is your feature request related to a problem? Please describe. For generic questions use Q&A section in the Discussions forum above.

For an input file `0_forward.fastq.gz` I get corrected reads `0_forward.fastq.00.0_0.cor.fastq.gz`. Wondering what the `00.0_0` part means. Is there any information encoded in this?

Also I get `0__unpaired.00.0_0.cor.fastq.gz` and was wondering why there is the double underscore?

For completeness: I'm running: `spades.py --meta --plasmid --only-error-correction -o 'output'  --pe-1 1 'paired_reads1/0_forward.fastq.gz' --pe-2 1 'paired_reads1/0_reverse.fastq.gz' --pe-or 1 fr`

**Comment by asl** [comment, marked as answer]
Hello

Since all the files are stored in the same directory the suffix is appended to make names unique. So, no "special" information encoded, just number of library / number of read file in that library. You can obtain the complete picture from the generated `corrected.yaml` file that describes the whole dataset in SPAdes-compatible format.

**Reply by bernt-matthias** [reply]
Thanks for the fast and very helpful reply.

**Comment by bernt-matthias** [comment]
May I ask two related questions:

1. what are the digits before the dot in  `00.0_0`? After the dot its library and read index separated by an underscore, as you said.
2. I had one call where the file was suddenly a fasta.gz file and the pattern was `0_0.cor.fasta.gz`, i.e. `.00.` was missing: the call was `spades.py --rnaviral  -o 'output'     --pe-1 1 'paired_reads1/0_1.fastq.gz' --pe-2 1 'paired_reads1/0_2.fastq.gz' --pe-or 1 fr --disable-rr --iontorrent   --phred-offset 33`

**Reply by asl** [reply]
I would suggest you not to reason about these numbers. This is just suffix added to made read names unique, that's it. The rnaviral cmdline also does look very strange :)

**Reply by bernt-matthias** [reply]
I'm currently working on the spades Galaxy wrappers https://github.com/galaxyproject/tools-iuc/pull/3904. I'm asking because for this irregularities in the generated output files names can be quite a hassle :) I'm doing this with regular expressions that are applied on the file names. For this it may be crucial to know which parts of the filename can be absent and which I can expect  always.

The command line is one that is generated by one of the tests in the Galaxy wrapper. What looks strange about the command line?

**Reply by asl** [reply]
> I'm doing this with regular expressions that are applied on the file names

For this particular task I would strongly suggest not to do this way. Instead – take the final dataset description YAML file, parse it and obtain all necessary information (libraries, which reads belong to which library, etc.)

> What looks strange about the command line?

Almost everything. E.g. phred offset makes sense only for read error correction, however rnaviral mode does not have one. Disabling repeat resolution is not advised as well :)

**Reply by bernt-matthias** [reply]
> take the final dataset description YAML file

Thanks for the suggestion. You mean: `corrected/corrected.yaml`?

> phred offset makes sense only for read error correction, however rnaviral mode does not have one

Then it seems strange that `corrected` contains data, isn't it?

Is there a way to know for which of spades' modes  read error correction is available (is it `--rna` and `--rnaviral`? For those the docs mention that  `--only-error-correction` is not available).

**Reply by asl** [reply]
> Then it seems strange that corrected contains data, isn't it?

Right. However, this might be an outcome of untested / unsupported combination of things here and there

> Is there a way to know for which of spades' modes read error correction is available 

Not automatically :) E.g. `--isolate` also implies it, etc.
